pandas/core/ops/__init__.py

```diff
@@ -756,9 +752,12 @@ def wrapper(self, other, axis=None):
                 )
 
         result = self._constructor(res_values, index=self.index)
-        # rename is needed in case res_name is None and result.name
-        #  is not.
-        return finalizer(result).rename(res_name)
+        result = finalizer(result)
+
+        # pin name for case where res_name is not and result.name is not
```

Comment:
```suggestion
        # Set the result's name after finalizer is called because finalizer would set it back to self.name
```

I think there was a missing None in the sentence, and AFAIU `result.name` is never "not None"

Is the above clear for you?